java/algorithm/src/main/java/org/apache/arrow/algorithm/search/VectorRangeSearcher.java

```diff
@@ -1,108 +1,108 @@
-
-import org.apache.arrow.algorithm.sort.VectorValueComparator;
-import org.apache.arrow.vector.ValueVector;
-
-/**
- * Search for the range of a particular element in the target vector.
- */
-public class VectorRangeSearcher {
-
-  /**
-   * Result returned when a search fails.
-   */
-  public static final int SEARCH_FAIL_RESULT = -1;
-
-  /**
-   * Search for the first occurrence of an element.
-   * The search is based on the binary search algorithm. So the target vector must be sorted.
-   * @param targetVector the vector from which to perform the search.
-   * @param comparator the criterion for the comparison.
-   * @param keyVector the vector containing the element to search.
-   * @param keyIndex the index of the search key in the key vector.
-   * @param <V> the vector type.
-   * @return the index of the first matched element if any, and -1 otherwise.
-   */
-  public static <V extends ValueVector> int getFirstMatch(
-          V targetVector, VectorValueComparator<V> comparator, V keyVector, int keyIndex) {
-    comparator.attachVectors(keyVector, targetVector);
-
-    int ret = SEARCH_FAIL_RESULT;
-
-    int low = 0;
-    int high = targetVector.getValueCount() - 1;
-
-    while (low <= high) {
-      int mid = low + (high - low) / 2;
-      int result = comparator.compare(keyIndex, mid);
-      if (result < 0) {
-        // the key is smaller
-        high = mid - 1;
-      } else if (result > 0) {
-        // the key is larger
-        low = mid + 1;
-      } else {
-        // an equal element is found
-        // continue to go left-ward
-        ret = mid;
-        high = mid - 1;
-      }
-    }
-    return ret;
-  }
-
-  /**
-   * Search for the last occurrence of an element.
-   * The search is based on the binary search algorithm. So the target vector must be sorted.
-   * @param targetVector the vector from which to perform the search.
-   * @param comparator the criterion for the comparison.
-   * @param keyVector the vector containing the element to search.
-   * @param keyIndex the index of the search key in the key vector.
-   * @param <V> the vector type.
-   * @return the index of the last matched element if any, and -1 otherwise.
-   */
-  public static <V extends ValueVector> int getLastMatch(
-          V targetVector, VectorValueComparator<V> comparator, V keyVector, int keyIndex) {
-    comparator.attachVectors(keyVector, targetVector);
-
-    int ret = SEARCH_FAIL_RESULT;
-
-    int low = 0;
-    int high = targetVector.getValueCount() - 1;
-
-    while (low <= high) {
-      int mid = low + (high - low) / 2;
-      int result = comparator.compare(keyIndex, mid);
-      if (result < 0) {
-        // the key is smaller
-        high = mid - 1;
-      } else if (result > 0) {
-        // the key is larger
-        low = mid + 1;
-      } else {
-        // an equal element is found,
-        // continue to go right-ward
-        ret = mid;
-        low = mid + 1;
-      }
-    }
-    return ret;
-  }
-}
+/*
```

Comment:
This change is a bit odd, I cannot see a formatting here, though locally the spotless gives a warning about the file. 

cc @danepitkin

Reply:
Good point, I did compare it with the main branch, it shows no change. That's the odd part. I need to check this again.

Reply:
\r\n -> \n

Reply:
@kou is exploring pre-commit (https://github.com/apache/arrow/issues/40417) which has checks for this sort of thing. Would Spotless be able to integrate with pre-commit?

Reply:
> \r\n -> \n
This is acceptable?

Reply:
@lidavidm I think we should be able to integrate it. We could create a bash script which does `mvn spotless:apply` and format the code. Shall we investigate in a separate PR?

Reply:
Issue created: https://github.com/apache/arrow/issues/40854

Reply:
The issue with this file are the line terminators (this is a Windows text file basically)